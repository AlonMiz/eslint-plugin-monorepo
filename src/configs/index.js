import * as recommended from './recommended';

export default {
  recommended,
};
