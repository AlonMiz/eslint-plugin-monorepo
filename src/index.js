export { default as rules } from './rules';
export { default as configs } from './configs';
