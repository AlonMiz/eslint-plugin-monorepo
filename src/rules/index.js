import * as noInternalImport from './no-internal-import';
import * as noRelativeImport from './no-relative-import';

export default {
  'no-internal-import': noInternalImport,
  'no-relative-import': noRelativeImport,
};
