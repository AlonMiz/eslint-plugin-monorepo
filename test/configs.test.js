import * as monorepo from '../src';

describe('configs', () => {
  test('exports an object', () => {
    expect(typeof monorepo.configs).toEqual('object');
  });

  test('matches snapshot', () => {
    expect(monorepo.configs).toMatchSnapshot();
  });
});
