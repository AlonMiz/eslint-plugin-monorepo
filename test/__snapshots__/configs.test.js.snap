// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configs matches snapshot 1`] = `
Object {
  "recommended": Object {
    "plugins": Array [
      "monorepo",
    ],
    "rules": Object {
      "monorepo/no-internal-import": "error",
      "monorepo/no-relative-import": "error",
    },
  },
}
`;
